### ISSUE TITLE
The parameter descriptions should distinguish between columns and column names. 

### ISSUE THREAD

**Issue by sfilipi** [dotnet/machinelearning#2177]
As @wschin pointed out here: https://github.com/dotnet/machinelearning/pull/2170#issuecomment-455346083 the parmaters, and in general our XML documentation should distinguish between columns and column names. 

This issue will be considered closed when all the params of this language:

```
/// <param name="labelColumn">The label column.</param>
/// <param name="featureColumn">The featureColumn column.</param>
/// <param name="weights">The optional weights column.</param>
```
Get changed to: 
```
/// <param name="labelColumnName">The name of the label column.</param>
/// <param name="featureColumnName">The name of the feature column.</param>
/// <param name="weightsColumnName">The name of the optional weights column.</param>
```

**Comment by glebuk** [comment]
Should the weights be renamed to weightsColumn, or perhaps exampleWeightsColumn?

**Comment by sfilipi** [comment]
> Should the weights be renamed to weightsColumn, or perhaps exampleWeightsColum

**Comment by HerraHak** [comment]
How can we contribute on this issue?

**Comment by abgoswam** [comment]
@sfilipi . 

For some learners, the feature column name is a `string`

https://github.com/dotnet/machinelearning/blob/4a71e501f09846974e550f817f3f8f1243054c1b/src/Microsoft.ML.HalLearners/HalLearnersCatalog.cs#L31-L34

But for some others its a `string[]`

https://github.com/dotnet/machinelearning/blob/4a71e501f09846974e550f817f3f8f1243054c1b/src/Microsoft.ML.StandardLearners/FactorizationMachine/FactorizationMachineCatalog.cs#L29-L32 

For consistency of API  

(1)  we will call both of them `featureColumnName`.  is that fine ?

(2) regarding order of arguments in API  should we adopt the following order between  

- featureColumnName, labelColumnName, weightColumnName
- featureColumnName, labelColumnName, groupIdColumnName, weightColumnName  (for algos that need group id)

@Ivanidzo4ka @glebuk

**Comment by abgoswam** [comment]
Assigning to myself  since this is superset of issues #2257, #2660 that i am fixing

**Comment by abgoswam** [comment]
Fixed by PR #2665

**Comment by wschin** [comment]
My feeling is that exampleWeightColumnName is a bit long, so #2873 is opened for further discussion.
